intro/scipy/index.rst

```diff
@@ -79,57 +79,55 @@ Scipy : high-level scientific computing
 
 File input/output: :mod:`scipy.io`
 ----------------------------------
+:mod:`scipy.io` contains functions for loading and saving data in
+several common formats including Matlab, IDL, Matrix Market, and
+Harwell-Boeing.
 
 **Matlab files**: Loading and saving::
 
     >>> import scipy as sp
     >>> a = np.ones((3, 3))
-    >>> sp.io.savemat('file.mat', {'a': a}) # savemat expects a dictionary
+    >>> sp.io.savemat('file.mat', {'a': a})  # savemat expects a dictionary
     >>> data = sp.io.loadmat('file.mat')
     >>> data['a']
     array([[1.,  1.,  1.],
            [1.,  1.,  1.],
            [1.,  1.,  1.]])
 
-.. warning:: **Python / Matlab mismatches**, *eg* matlab does not represent 1D arrays
+.. warning:: **Python / Matlab mismatch**: The Matlab file format does not support 1D arrays.
 
    ::
 
       >>> a = np.ones(3)
       >>> a
       array([1.,  1.,  1.])
+      >>> a.shape
+      (3,)
       >>> sp.io.savemat('file.mat', {'a': a})
-      >>> sp.io.loadmat('file.mat')['a']
+      >>> a2 = sp.io.loadmat('file.mat')['a']
+      >>> a2
       array([[1.,  1.,  1.]])
+      >>> a2.shape
+      (1, 3)
 
-   Notice the difference?
+   Notice that the original array was a one-dimensional array, whereas the
+   saved and reloaded array is a two-dimensional array with a single row.
 
-|
-
-.. Comments to make doctests pass which require an image
-    >>> import matplotlib.pyplot as plt
-    >>> plt.imsave('fname.png', np.array([[0]]))
-
-**Image files**: Reading images::
-
-    >>> import imageio.v3 as iio
-    >>> iio.imread('fname.png')
-    array([[[ 0,   0,  0, 255]]], dtype=uint8)
-    >>> # Matplotlib also has a similar function
-    >>> import matplotlib.pyplot as plt
-    >>> plt.imread('fname.png')
-    array([[[0., 0., 0., 1.]]], dtype=float32)
+   For other formats, see the :mod:`scipy.io` documentation.
```

Comment:
Do we want to conclude each section with a line like this, or should the section just end?

Reply:
Not sure it is necessary, but seems fine to me.